core/api/service/impl/api_service_impl.cpp

```diff
@@ -515,7 +515,7 @@ namespace kagome::api {
   void ApiServiceImpl::onStorageEvent(SubscriptionSetId set_id,
                                       SessionPtr &session,
                                       const Buffer &key,
-                                      const Buffer &data,
+                                      const std::optional<Buffer> &data,
```

Comment:
Hm, I think when it's not none, we'll be copying the buffer here, if in the calling code it's not already in an optional, which is not nice.